### DISCUSSION TITLE
How can we ensure unique records when syncing inventory ledger reports?

### DISCUSSION THREAD

**Discussion by ProNotion** [amzn/selling-partner-api-models#4321] [Q&A]
The inventory ledger does not come with any unique record identifiers so when trying to sync it locally we cannot easily ensure we do not duplicate records. Is there a specific reason for this? Is there any recommendation on how we can safely sync it?

**Comment by ritikacd** [comment]
Hi @ProNotion 

Please review the Inventory ledger report's behaviors and schedule report using one of the report options available [here](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba#inventory-ledger-report---summary-view). Based on provided report options, and date time period, a report will be generated for that time only. You can use ASIN or SKU to see individual inventory balances. If you are only fetching inventory data using SP-API, and retrieving report per schedule only once, is there a reason you will see duplicate records? 

Thanks,
Ritika
Selling Partner API Developer Services

**Reply by phstc** [reply]
> per schedule only once

@ritikacd, depending on the process, it may be re-executed on failure or by accident, which can cause duplicates. Ensuring once per schedule can be error-prone. Is there any other identifier we could use? I'm asking particularly about the event type Receipts. I wonder if we could generate a fingerprint of the whole row as a unique identifier. 

@ProNotion did you find any reliable solution for ensuring unique records?

**Reply by ritikacd** [reply]
Hi @phstc, can you please share which Inventory Ledger Report (Summary view or Detailed View) is being used?

**Reply by phstc** [reply]
Hi @ritikacd I'm using the detailed view.

My current workaround for Receipts is to convert the entire row into a fingerprint (MD5) and use it as an identifier to prevent duplicates. 

I can't use the whole row for Adjustments because reconciled and unreconciled quantities can change overtime.

**Reply by ritikacd** [reply]
Hi @phstc The detailed report includes every inventory movement, so you may see multiple rows for the same ASIN/SKU if there have been multiple activities. However, there should not be duplicate entries. If you’re seeing duplicates, could you please share an example so we can investigate?
